_posts/linguistic-limitation.md

```diff
@@ -0,0 +1,82 @@
+---
+layout: post
+header_image: /img/posts/linguistic-limitation/header.jpg
+title: "A linguistic limitation"
+author: Mauro Servienti
+synopsis: "The language we use is crucial to get the right message through. The words we use have to be as specific as we can to avoid misunderstandings. It seems we have space for improvement, and we should be doing a better job."
+tags:
+- ubiquitous language
+---
+
+I have a feeling this is going to be a short one. I have no answers, only doubts.
+
+A couple of weeks ago, I was chatting with a friend, discussing some architectural designs. At a certain point, we mentioned publishing an event as a solution to one of the issues. All of a sudden, we stopped and asked the question: what kind of event? A domain event or a pub/sub-one?
+
+As a result, we stopped discussing the initial architectural concern and moved to a completely different topic: the ubiquitous language or the lack of a proper one.
+
+Martin Fowler [defines the ubiquitous language](https://www.martinfowler.com/bliki/UbiquitousLanguage.html) as follows:
+
+> Ubiquitous Language is the term Eric Evans uses in Domain Driven Design for the practice of building up a common, *rigorous language between developers and users*. This language should be based on the Domain Model used in the software - hence the need for it to be rigorous, since software doesn't cope well with ambiguity.
+
+(emphasis is mine)
+
+We were not in the context of a discussion between developers and users, and we were not discussing any business concerns. The sentence that triggered the follow-up discussion was more or less the following:
+
+> The endpoint could initiate the process by raising an event. It'll make the background processing asynchronous.
+
+I immediately see three points of confusion. What's an endpoint?
+
+A dictionary [defines an endpoint as](https://www.lexico.com/en/definition/endpoint):
+
+> The final stage of a period or process.
+
+And it provides additional definitions for chemistry and mathematics. There's nothing about IT. So, is the endpoint mentioned above an HTTP endpoint or something else?
```

Comment:
I prefer to avoid use of "The dictionary defines X as ..." since no one is really interested in the dictionary definition. Instead, consider:

```suggestion
I immediately see three points of confusion. What's an endpoint? Is it an HTTP endpoint, a ReST endpoint, or something else entirely?
```